lib/hanami/action/flash.rb

```diff
@@ -125,94 +133,124 @@ def inspect
       #
       # @since 0.3.0
       # @api private
-      def flash
+      def _data
         @session[SESSION_KEY] || {}
       end
 
-      # The flash registry that holds the data **only for** the current request
+      # Remove the flash entirely from the session if empty.
       #
-      # @return [Hash] the flash for the current request
+      # @return [void]
       #
       # @since 0.3.0
       # @api private
-      def data
-        flash[@request_id] || {}
+      #
+      # @see Hanami::Action::Flash#empty?
+      def remove!
```

Comment:
Any reason for this being a *bang* methods? I mean, in Ruby's stdlib bang methods are destructive methods and they only exist if a non-banged variant is even possible. Are there any better names that do not include a bang?

Not really a huge problem, but having different semantics across each piece of the framework + language, IMO, kinda break the POLS.